targets/mbedos5/tools/generate_pins.py

```diff
@@ -53,41 +69,50 @@ def find_file(root_dir, directories, name):
 
     for root, dirs, files in os.walk(root_dir, topdown=True):
         # modify dirs in place
-        dirs[:] = filter(lambda x: x in directories or not x.startswith('TARGET_'), dirs)
+        dirs[:] = [directory for directory in dirs if directory in directories or not directory.startswith('TARGET_')]
 
         if name in files:
             return os.path.join(root, name)
 
+
 def enumerate_includes(root_dir, directories):
     """
     Walk through the directory tree, starting at root_dir, and enumerate all
     valid include directories.
     """
-    for root, dirs, files in os.walk(root_dir, topdown=True):
+    for root, dirs, _ in os.walk(root_dir, topdown=True):
         # modify dirs in place
-        dirs[:] = filter(lambda x: x in directory_labels
-                                or (    not x.startswith('TARGET_')
-                                    and not x.startswith('TOOLCHAIN_')), dirs)
+        dirs[:] = [dir_label for dir_label in dirs
+                   if dir_label in directories
+                   or (not dir_label.startswith('TARGET_')
+                       and not dir_label.startswith('TOOLCHAIN_'))]
         yield root
 
 
 class TypeDeclVisitor(c_ast.NodeVisitor):
-    def __init__(self, filter_names=[]):
+    """
+    A TypeDecl visitor class that walks the ast and calls a visitor function for every node found.
+    """
+    def __init__(self, filter_names=None):
+        filter_names = filter_names or []
```

Comment:
this two-step assignment is unnecessary. let's just use `self.names = filter_names or []`